src/Cli/dotnet/MSBuildOptions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Linq;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class MSBuildOptions
```

Comment:
My worry is we need to keep this list in sync. If msbuild adds a new option, and we forget to add it here, it would have something weird right?

We don't have to manually add this list to before. It is because of we don't have the feature of "unix combine"(forget the proper name for things like `-fxd`) It is a good feature for system.commandline but I think not to use this combining feature in SDK is fine. 1. We don't have the feature before. 2. And we don't have too many one letter options that worth combining.

Reply:
Yeah, I agree that if we could disable the unix combine feature that would be ideal but right now that's not a System.CommandLine option. Instead, we compromised by making a change to ensure that the longer option is matched when options prefix each other (for example `-c` and `-clp` used to match `-c` with the argument `lp` and now it matches `-clp`). 

We don't actually need most of these options, I just included all of them here for completeness. Really, we only need to include those options that are prefixed by built in .NET options (such as `-clp` since it's prefixed by `-c` and `-outputResultsCache` since it's prefixed by `-o`). If you think it would be better to only include the fully necessary options, I'll probably be able to remove about half of these.

Reply:
The problem is long term maintenance. When msbuild adding a new option, people would not remember to see if it happens to match a SDK option prefix. So it will be a bug down the road every single time.

Reply:
Yes, I understand and definitely agree that it's a maintenance problem. I'm not sure how to avoid this though, it sounded like @jonsequitur wasn't willing to add a System.CommandLine option to fully opt out of the unix combine feature, is that right?

Reply:
> wasn't willing to add a System.CommandLine option to fully opt out of the unix combine feature, is that right?

Could you point me to the thread?

Reply:
@jonsequitur and I discussed over teams.